src/Pages/ProductDetailPage/index.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect } from "react";
+import { Product } from "../../Types/types";
+import { Grid, Typography } from "@mui/material";
+import { useParams } from "react-router-dom";
+import useGlamGrabStore from "../../store/store";
+import { ContainerWithTitle } from "../../components/PageLayout";
+import ButtonAppBar from "../../components/Header";
+import { MarginMediumPx, MarginSmallPx } from "../../utils/styles";
+import ActionBar from "../../components/ProductCard/productActionBar";
+import pricePerUnitWithCurrency from "../../utils/helper";
+
+const ProductDetailPage: React.FC = () => {
+  const { productId } = useParams();
+  const { products, fetchAllProducts, productsInBag } = useGlamGrabStore();
+  const product = products.filter((p: Product) => p.id == Number(productId))[0];
+  const productInBag = productsInBag.filter(
+    (p: Product) => p.id == Number(productId)
+  )[0];
+  const addToBag = useGlamGrabStore((state) => state.addOneToBag);
+  const removeFromBag = useGlamGrabStore((state) => state.removeOneFromBag);
+  useEffect(() => {
+    fetchAllProducts();
+  }, [fetchAllProducts]);
+  console.log(products);
+  console.log(product);
+  return (
+    <>
+      <ButtonAppBar />
+      <ContainerWithTitle title="Product Detail">
+        {product && (
+          <Grid
+            container
+            sx={{
```

Comment:
in my team, you would be directly asked to create a separate CSS file (or loess, scss) to put your classnames in it and not to use Style in the JSX